src/vite.rs

```diff
@@ -229,8 +235,119 @@ impl std::error::Error for ViteError {
+            .port(8080)
+            .main(main_script)
+            .lang("lang-id")
+            .title("app-title-here")
+            .react();
+
+        let config = development.into_config();
+        assert_eq!(config.version(), None);
+
+        let config_layout = config.layout();
+        let binding = config_layout(r#"{"someprops": "somevalues"}"#.to_string());
+        let rendered_layout = binding.as_str();
+        println!("{}", rendered_layout);
+        assert!(rendered_layout.contains("<html lang=\"lang-id\">"));
+        assert!(rendered_layout.contains("<title>app-title-here</title>"));
+        assert!(rendered_layout.contains(r#"{&quot;someprops&quot;: &quot;somevalues&quot;}"#));
+        assert!(rendered_layout.contains("http://localhost:8080/@vite/client"));
+        assert!(rendered_layout.contains("window.__vite_plugin_react_preamble_installed__ = true"));
+    }
+
+    #[test]
+    fn test_production_new_entry_missing() {
+        let result = Production::new(r#"{"main.js": {}}"#, "nonexistent.js");
+        assert!(matches!(result, Err(_)));
+    }
+
+    #[test]
+    fn test_production_new() {
+        let manifest_content =
+            r#"{"main.js": {"file": "main.hash-id-here.js", "css": ["style.css"]}}"#;
+        let production_res = Production::new(manifest_content, "main.js");
+        assert!(production_res.is_ok());
+
+        let production = production_res.unwrap();
+        let content_hash = encode(Sha1::digest(manifest_content.as_bytes()));
+        assert_eq!(production.main.css, Some(vec!(String::from("style.css"))));
+        assert_eq!(production.title, "Vite");
+        assert_eq!(production.main.file, "main.hash-id-here.js");
+        assert_eq!(production.main.integrity, None);
+        assert_eq!(production.lang, "en");
+        assert_eq!(production.version, content_hash);
+    }
+
+    #[test]
+    fn test_production_builder_methods() {
+        let production = Production::new(
+            r#"{"main.js": {"file": "main.hash-id-here.js", "css": ["style.css"]}}"#,
+            "main.js",
+        )
+        .unwrap()
+        .lang("fr")
+        .title("Untitled Axum Inertia App");
+
+        assert_eq!(production.lang, "fr");
+        assert_eq!(production.title, "Untitled Axum Inertia App");
+    }
+
+    #[test]
+    fn test_production_into_config() {
+        let production = Production::new(r#"{"main.js": {"file": "main.hash-id-here.js", "integrity": "sha000-shaHashHere1234", "css": ["style.css"]}}"#, "main.js").unwrap()
+            .lang("jv")
+            .title("Untitled Axum Inertia App");
+
+        let config = production.into_config();
+        let config_layout = config.layout();
+        let binding = config_layout(r#"{"someprops": "somevalues"}"#.to_string());
+        let rendered_layout = binding.as_str();
+
+        assert!(rendered_layout.contains("<script type=\"module\" src=\"/main.hash-id-here.js\" integrity=\"sha000-shaHashHere1234\"></script>"));
```

Comment:
awesome. love this test.

one possible small improvement: can we test that without a provided `integrity` hash, that we leave off the `integrity` attribute on the script tag? I think we may need to use an `@if let...` in the html layout above, matching on the optional integrity hash. [Here's some relevant documentation](https://maud.lambda.xyz/control-structures.html).